### PR TITLE
Compaction::IsTrivialMove relaxing

### PR DESCRIPTION
**Pull request by artemy-kolesnikov**
IsTrivialMove returns true if no input file overlaps with output_level+1 with more than max_compaction_bytes_ bytes.

### CODE REVIEW

File: db/compaction.cc [hunk 1]
```diff
@@ -232,24 +232,41 @@ bool Compaction::IsTrivialMove() const {
...
-          InputCompressionMatchesOutput() &&
-          TotalFileSize(grandparents_) <= max_compaction_bytes_);
+          InputCompressionMatchesOutput())) {
+    return false;
+  }
+
+  // assert inputs_.size() == 1
+
+  for (const auto& file : inputs_.front().files) {
+    std::vector<FileMetaData *> fileGrandparents;
```
Comment (siying) [inline db/compaction.cc]: Naming convention is `file_grand_parents`.

File: db/compaction.h [hunk 2]
```diff
@@ -258,6 +258,8 @@ class Compaction {
   static bool IsFullCompaction(VersionStorageInfo* vstorage,
                                const std::vector<CompactionInputFiles>& inputs);
 
+  VersionStorageInfo* vstorage_;
```
Comment (siying) [inline db/compaction.h]: Maybe call it `input_vstorage_`.

Replace all the input_version_->storage_info() with this one.

File: db/compaction_picker_test.cc [hunk 3]
```diff
@@ -1177,6 +1177,50 @@ TEST_F(CompactionPickerTest, MaxCompactionBytesNotHit) {
...
+
+  Add(3, 5U, "120", "130", 6000U);
+  Add(3, 6U, "140", "150", 6000U);
+  UpdateVersionStorageInfo();
+
+  std::unique_ptr<Compaction> compaction(level_compaction_picker.PickCompaction(
+    cf_name_, mutable_cf_options_, vstorage_.get(), &log_buffer_));
+  ASSERT_TRUE(compaction.get() != nullptr);
+  ASSERT_FALSE(compaction->IsTrivialMove());
+}
```
Comment (siying) [inline db/compaction_picker_test.cc]: Should add a unit test that determines it is not trivial move because of max_compaction_bytes.

Comment (artemy-kolesnikov) [inline db/compaction_picker_test.cc]: There is test for this - IsTrivialMoveOff

Comment (siying) [inline db/compaction_picker_test.cc]: It's fine then.

**Comment by yiwu-arbug** [comment]
Let me try to fix the transaction lock related warnings. After that let's rebase and rerun tests. You can also run `make format` to format the code.

**Comment by yiwu-arbug** [comment]
@artemy-kolesnikov mind rebase to trigger tests again?

**Comment by artemy-kolesnikov** [comment]
Have rebased. Tests should rerun.

File: db/compaction.cc [hunk 4]
```diff
@@ -226,30 +226,48 @@ bool Compaction::IsTrivialMove() const {
...
+  // assert inputs_.size() == 1
+
+  for (const auto& file : inputs_.front().files) {
+    std::vector<FileMetaData *> file_grand_parents;
+    if (output_level_ + 1 >= number_levels_) {
+      continue;
+    }
+    input_vstorage_->GetOverlappingInputs(output_level_ + 1, &file->smallest,
+                                    &file->largest, &file_grand_parents);
+    const auto compaction_size = file->fd.GetFileSize() + TotalFileSize(file_grand_parents);
```
Comment (siying) [inline db/compaction.cc]: Is the 80-char rule followed here?

If you run on dev servers, you can run "make format" and it will fix some issues for you.

**Comment by yiwu-arbug** [comment]
Test failures are not related to the change. Landing.

**Pull request closed without merging**